### ISSUE TITLE
UI Scaffolding

### ISSUE THREAD

**Issue by zak10** [auto-profile/ui#1]
Start with https://github.com/rafaelhz/react-material-admin-template

**Comment by piotrromanowski** [comment]
Any opposition to mooving to/using https://github.com/facebookincubator/create-react-app instead while still in early development  ?

**Comment by zak10** [comment]
@piotrromanowski I'm in for that. @davewalk thoughts?

**Comment by davewalk** [comment]
@piotrromanowski I don't know much about React - why would you say that template would be a better way to start?

**Comment by piotrromanowski** [comment]
I'm not saying it's the best template, however some selling points
- Huge community
- No configuration needed
- Flow Type support
- Testing with Jest built in
- Ejecting ability

I don't use this particular template, however "it's all the hype" in the react community right now and suggested by many. Up to you guys! :smile:

**Comment by zak10** [comment]
All aboard the hype train 🚋 Let's do it

**Comment by davewalk** [comment]
Sounds good, I'm in. Do you want to send a PR for the basic React scaffolding to get us going?

**Comment by piotrromanowski** [comment]
sure! Will do

**Comment by piotrromanowski** [comment]
hmm, on the other hand create-react-app doesn't support hot reloading which is really nice during development. Thoughts? Maybe just stick it out with this template and add any features that create react app uses to this project?

**Comment by piotrromanowski** [comment]
Going to give it a shot now. I'll see if it integrates with the go server as well

**Comment by piotrromanowski** [comment]
yep, just as I had suspected. So create-react-app doesn't have hot-reloading like this template, but does have live-reloading. But the bigger issue is that it's not as trivial to integrate it with the go server as it is now. In order to use create-react-app here, the client and server would need to be completely separate and the server would only be used as a REST api.
Is this current project a clone of some template? If it's not too much of a burden, I would suggest splitting up the project to create-react-app client, and go REST api. So when running the app you would start the create-react-app server which serves on localhost:3000 and the REST api on another port. This would then require some code to allow [CORS.](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS) When it comes to building the app for prod, you would serve the static files through the go server.
Or, just continue with the current setup and maybe next add a test suite/runner for the client.

**Comment by zak10** [comment]
That's what we were thinking initially. I think as long as we can maintain an easy way to bring up both dev and prod environments, having separate projects for UI and API are fine. my 2c.
